src/community/wmts-multi-dimensional/src/main/java/org/geoserver/gwc/wmts/MultiDimensionalExtension.java

```diff
@@ -162,15 +175,44 @@ private Domains getDomains(SimpleConveyor conveyor) throws Exception {
             boundingBox = new ReferencedEnvelope(boundingBox, CRS.decode(gridSubset.getSRS().toString()));
         }
         // add any domain provided restriction and set the bounding box
-        Filter filter = Filter.INCLUDE;
+        ResourceInfo resource = layerInfo.getResource();
+        Filter filter = DimensionsUtils.getBoundingBoxFilter(resource, boundingBox, filterFactory);
         for (Dimension dimension : dimensions) {
             Object restriction = conveyor.getParameter(dimension.getDimensionName(), false);
-            dimension.setBoundingBox(boundingBox);
-            dimension.addDomainRestriction(restriction);
-            filter = filterFactory.and(filter, dimension.getFilter());
+            if (restriction != null) {
+                Tuple<String, String> attributes = DimensionsUtils.getAttributes(resource, dimension);
+                filter = appendDomainRestrictionsFilter(filter, attributes.first, attributes.second, restriction);
+            }
         }
+        // compute the bounding box
+        ReferencedEnvelope spatialDomain = null;
+        if (requestedDomains == ALL_DOMAINS || requestedDomains.contains(SPACE_DIMENSION)) {
+            spatialDomain = DimensionsUtils.getBounds(resource, filter);  
+        } 
         // encode the domains
-        return new Domains(dimensions, layerInfo, boundingBox, SimplifyingFilterVisitor.simplify(filter));
+        return new Domains(dimensions, layerInfo, spatialDomain, SimplifyingFilterVisitor.simplify(filter));
+    }
+
+    
+
+    /**
+     * Helper method that will build a dimension domain values filter based on this dimension start and end
+     * attributes. The created filter will be merged with the provided filter.
+     */
+    protected Filter appendDomainRestrictionsFilter(Filter filter, String startAttribute, String endAttribute, Object domainRestrictions) {
+        DimensionFilterBuilder dimensionFilterBuilder = new DimensionFilterBuilder(filterFactory);
+        List<Object> restrictionList = domainRestrictions instanceof Collection ? new ArrayList<>((Collection) domainRestrictions) : Arrays.asList(domainRestrictions);
+        dimensionFilterBuilder.appendFilters(startAttribute, endAttribute, restrictionList);
+        return filterFactory.and(filter, dimensionFilterBuilder.getFilter());
+    }
+
+    private Set<String> getRequestedDomains(Object domains) {
+        if (domains == null) {
+            return ALL_DOMAINS;
+        }
+
+        String[] domainNames = domains.toString().trim().split("\\s*,\\s*");
```

Comment:
If ever domains are an empty string ``""``, the split will return an array with an empty string element, is it worth to add also an empty check in the if above ?

Reply:
I don't believe a nameless domain is allowed, but the user can ask for it... instead of an empty test, the code should probably have an existence check (which is indeed missing now) so that the client would get back an error when requesting a non existing dimension.